05-pagination-querying.md

```diff
@@ -0,0 +1,366 @@
+be overridden by specifying the sorting order in the following manner:
+
+Request
+
+`GET /cars?sort=make,desc`
+
+Response
+
+~~~~
+{
+	"content": [
+	  {
+		"make": "Volkswagen",
+		"model": "Amarok"
+	  },
+	  {
+		"make": "Volkswagen",
+		"model": "Beetle"
+	  },
+	  {
+		"make": "Volkswagen",
+		"model": "Golf"
+   	  },
+   	  {
+		"make": "Volkswagen",
+		"model": "Polo"
+	  },
+	  {
+		"make": "Volkswagen",
+		"model": "Santana"
+	  }
+	],
+	"number": 0,
+	"size": 5,
+	"first": true,
+	"last": false,
+	"totalElements": 100,
+    "totalPages": 20,
+    "sort": [{
+        "direction": "DESC",
+        "property": "make"
+    }]
+}
+~~~~
+
+The sorting on 'make' is now done in a descending order.
+
+If you want you can also specify multiple 'sort' parameters. They are processed in order, for example:
+
+Request
+`GET /cars?sort=make&sort=model,desc`
+
+Would first order the elements by make, and then by descending model.
+
+Response:
+~~~~
+{
+	"content": [
+	  {
+		"make": "Audi",
+		"model": "Q7"
+	  },
+	  {
+		"make": "Audi",
+		"model": "Q6"
+	  },
+	  {
+		"make": "Audi",
+		"model": "Q5"
+   	  },
+   	  {
+		"make": "Audi",
+		"model": "Q4"
+	  },
+	  {
+		"make": "Audi",
+		"model": "Q3"
+	  }
+	],
+	"number": 0,
+	"size": 5,
+	"first": true,
+	"last": false,
+	"totalElements": 100,
+    "totalPages": 20,
+    "sort": [
+      {
+        "property": "make",
+        "direction": "ASC"
+      },
+      {
+        "property": "model,
+        "direction": "DESC"
+      }
+    ]
+}
+~~~~
+
+## Querying ##
+
+Most of the time the user is not interested to browse through a large paginated
+list of all data present on the server. To allow the user to find information he/she is looking for,
+it is possible to provide query parameters to narrow the results.
+
+For example, if our Car resource has a make and a model and the user is only interested
+in finding Hyundai cars, the following request is be make:
+
+Request:
+
+`GET /cars?make=hyundai`
+
+Response:
+
+~~~~
+{
+	"content": [
+	  {
+		"make": "Hyundai",
+		"model": "Accent"
+	  },
+	  {
+		"make": "Hyundai",
+		"model": "Aero"
+	  },
+	  {
+		"make": "Hyundai",
+		"model": "Atos"
+   	  },
+   	  {
+		"make": "Hyundai",
+		"model": "Azera"
+	  },
+	  {
+		"make": "Hyundai",
+		"model": "Chorus"
+	  }
+	],
+	"number": 0,
+	"size": 5,
+	"first": true,
+	"last": false,
+	"totalElements": 10,
+	"totalPages": 2,
+}
+~~~~
+
+The user can further narrow the search by applying more query parameters. For example:
+
+`GET /cars?make=hyundai&model=a`
+
+Could return all Hyundai cars for which the model name starts with an 'a'.
+
+## Combining pagination, querying and sorting ##
+The parameters to apply pagination, querying and sorting mentioned in the previous sections of this chapter
+can be combined.
+
+For example using the following request:
+
+`GET /cars?page=1&make=hyundai&sort=model,desc`
+
+Would return the second page of Hyundai cars and sort them by model in descending order.
```

Comment:
I would also like to see an example were we want to see all 'hyundai' and 'audi's. What do the query parameters look like then.